core/src/test/java/com/radixdlt/api/core/TransactionStreamTest.java

```diff
@@ -331,15 +333,22 @@ public void test_previous_state_identifiers_and_proofs() throws Exception {
                       .fromStateVersion(1L));
       assertThat(firstPartResponse.getProofs()).isNull();
 
-      var firstPartResponseWithProofs =
-          getStreamApi()
-              .streamTransactionsPost(
-                  new StreamTransactionsRequest()
-                      .network(networkLogicalName)
-                      .includeProofs(true)
-                      .limit(100)
-                      .fromStateVersion(1L));
-      assertThat(firstPartResponseWithProofs.getProofs().size()).isEqualTo(13);
+      // An async "proofs pruning" is supposed to run over the genesis' epoch...
+      Awaitility.await()
+          .atMost(2, TimeUnit.SECONDS)
+          .untilAsserted(
+              () -> {
+                var firstPartResponseWithProofs =
+                    getStreamApi()
+                        .streamTransactionsPost(
+                            new StreamTransactionsRequest()
+                                .network(networkLogicalName)
+                                .includeProofs(true)
+                                .limit(100)
+                                .fromStateVersion(1L));
+                // ... leaving 9 (out of original 13) ledger proofs.
```

Comment:
Why are 9 proofs kept instead of all of them being deleted? Are epochs short and these are the end of epoch proofs?

In any case, this assertion feels too specific, I'd rather we just assert some proofs are returned, or that the max gap between proofs is 1000 or something :)

Reply:
> Why are 9 proofs kept instead of all of them being deleted?

According to logs in this tests, we are at epoch 2, round 8. There are 8 proofs in epoch 2 (the GC never run over epoch 2, because it is not a concluded epoch), and a single proof from epoch 1 (i.e. only 1 is retained from original 5).

In general, this test is not about ledger proofs GC (and not even about ledger sync limits), so the best way to deal with confusion here is to relax the asserts (just "some proofs are returned") so that we don't have to consider the async pruning.